hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -200,6 +203,19 @@
       .defaultValue("true")
       .withDocumentation("Similar to " + BLOOM_INDEX_UPDATE_PARTITION_PATH_ENABLE + ", but for simple index.");
 
+  // ***** Bucket Index Configs *****
+  public static final ConfigProperty<Integer> BUCKET_INDEX_BUCKET_NUM = ConfigProperty
+      .key("hoodie.index.bucket.num")
+      .defaultValue(-1)
+      .withDocumentation("Only applies if index type is BUCKET_INDEX. Determine the bucket num of the hudi table, "
```

Comment:
Just. `Determine the total number of buckets in each table partition.` ?

Reply:
I has already implemented partition-level bucket, but not included in this pr. Besides, the partition-level bucket optimization rules is introduced to Spark for efficient reading. Both of them will be contributed to community in the following prs